src/sentry/integrations/github/client.py

```diff
@@ -19,12 +19,12 @@ def get_last_commits(self, repo, end_sha):
         # return api request that fetches last ~30 commits
         # see https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository
         # using end_sha as parameter
-        return self.get(u"/repos/{}/commits".format(repo), params={"sha": end_sha})
+        return self.get_cached(u"/repos/{}/commits".format(repo), params={"sha": end_sha})
 
     def compare_commits(self, repo, start_sha, end_sha):
         # see https://developer.github.com/v3/repos/commits/#compare-two-commits
         # where start sha is oldest and end is most recent
-        return self.get(u"/repos/{}/compare/{}...{}".format(repo, start_sha, end_sha))
+        return self.get_cached(u"/repos/{}/compare/{}...{}".format(repo, start_sha, end_sha))
 
     def repo_hooks(self, repo):
         return self.get(u"/repos/{}/hooks".format(repo))
```

Comment:
out of curiosity why aren't these cached too?

Reply:
@wmak Great question. There are two main reasons:
1. This endpoint doesn't show up in our logs for the `You have triggered an abuse detection mechanism` error which is why we need caching
2. The repo hooks might change for a repo. I wanted to avoid caching endpoints that could change if I could avoid it. On the other hand, the response values of endpoints that refer to a specific commit hash should not change (or at least I don't think they should change). Commits are immutable so the commits up to a particular sha should never change.